libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1079,7 +1089,26 @@ MQTTStatus_t MQTT_SerializeDisconnect( const MQTTFixedBuffer_t * const pBuffer )
 
 MQTTStatus_t MQTT_SerializePingreq( const MQTTFixedBuffer_t * const pBuffer )
 {
-    return MQTTSuccess;
+    MQTTStatus_t status = MQTTSuccess;
+
+    if( pBuffer == NULL )
+    {
+        IotLogError( "pBuffer is NULL. " );
+        status = MQTTBadParameter;
+    }
+    else if( pBuffer->size < MQTT_PACKET_PINGREQ_SIZE )
+    {
+        IotLogError( "Insufficient memory for packet." );
```

Comment:
This log message can be improved to include the provided and expected buffer size.